sys/event/event.c

```diff
@@ -59,13 +93,18 @@ event_t *event_get(event_queue_t *queue)
 
 event_t *event_wait(event_queue_t *queue)
 {
-    thread_flags_wait_any(THREAD_FLAG_EVENT);
-    unsigned state = irq_disable();
-    event_t *result = (event_t *) clist_lpop(&queue->event_list);
-    if (clist_rpeek(&queue->event_list)) {
-        queue->waiter->flags |= THREAD_FLAG_EVENT;
-    }
-    irq_restore(state);
+    assert(queue);
+    event_t *result;
+
+    do {
+        unsigned state = irq_disable();
```

Comment:
sadly it would not help, as claiming the queue and actually processing events is not the same thing (i.e. they might be invoked at different times).  

> I don't like this loop much.

I thought so :-). But in the end it does not introduce any runtime/code size penalties while giving the additional wanted functionality, so I'd say it suits the purpose here.

Reply:
ok!